controllers/core_monitoring_controller.go

```diff
@@ -60,5 +65,30 @@ func (r *CoreMonitoringReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&corev1.ConfigMap{}).
 		Owns(&corev1.Secret{}).
 		Owns(&corev1.Service{}).
+		Watches(
+			&source.Kind{Type: &corev1beta1.Gateway{}},
+			handler.EnqueueRequestsFromMapFunc(r.findGateway),
+			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
+		).
 		Complete(r)
 }
+
+func (r *CoreMonitoringReconciler) findGateway(o client.Object) []ctrl.Request {
+	monitoringClusters := &corev1beta1.MonitoringClusterList{}
```

Comment:
If we're looking up `MonitoringClusters` wouldn't this be better of called `findMonitoringClusters` ?

Reply:
good catch